### PR TITLE
Upgrade test infrastructure.

### PR DESCRIPTION
**Pull request by karelmaxa**
This PR introduces a brand new testing infrastructure that enables automated test execution and verification of test results.

### CODE REVIEW

File: .common.sh [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+exec_db() {
+  docker exec -i wrenidm-test-db "$@"
+}
+
+exec_ldap() {
+  docker exec -i wrenidm-test-ldap "$@" < /dev/stdin
+}
+
+delete_managed() {
```
Comment (pavelhoral) [inline .common.sh]: Is this really necessary?

File: docker/wrenidm/conf/config.properties [hunk 2]
```diff
@@ -93,7 +93,7 @@ org.osgi.service.http.enabled=true
 org.osgi.service.http.secure.enabled=true
 
 # Disable file configuration update
-felix.fileinstall.enableConfigSave=false
+felix.fileinstall.disableConfigSave=false
```
Comment (pavelhoral) [inline docker/wrenidm/conf/config.properties:96]: True?

File: sync/01-sync.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  --connect-to "wrenidm.wrensecurity.local:80:10.0.0.11:8080" \
+  "http://wrenidm.wrensecurity.local/openidm/managed/user?_queryId=query-all-ids" \
+| assert_response_status \
+| assert_response_body '.resultCount == 2' \
+| assert_response_body '.result[0]._id | test("^sync(1|2)$")' \
+| assert_response_body '.result[1]._id | test("^sync(1|2)$")' \
+> /dev/null
+
+# 4. Check accounts in LDAP
+sleep 3 # Ensure that implicit synchronization has been performed
```
Comment (pavelhoral) [inline sync/01-sync.sh]: This is quite suboptimal... this should be while loop or inotify on logs that checks for a message.

File: sync/02-cleanup.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+. "$(dirname "${BASH_SOURCE[0]}")/../.common.sh"
+. "$(dirname "${BASH_SOURCE[0]}")/../.client.sh"
+
+log_message "02-cleanup.sh..."
+
+delete_managed "user"
```
Comment (pavelhoral) [inline sync/02-cleanup.sh]: I am not a fan of this type of cleanup. Would be better if the test uses it's own data and maybe cleans it up during init phase.

**Review by pavelhoral — CHANGES_REQUESTED** [review]

File: docker/wrenidm/conf/config.properties (hunk 2 above)
Comment (karelmaxa) [reply docker/wrenidm/conf/config.properties:96]: No, the value `false` is correct according to the current implementation. See [JSONConfigInstaller.java#L145](https://github.com/WrenSecurity/wrenidm/blob/c84d0e59c7ad52195d4fde1dcc2b9ce95c674b5c/openidm-config/src/main/java/org/forgerock/openidm/config/installer/JSONConfigInstaller.java#L145). 

But you're right, it doesn't make sense. We should change it in the Wren:IDM.

File: .common.sh (hunk 1 above)
Comment (karelmaxa) [reply .common.sh]: Actually not, I have restricted the search queries so the cleanup is no longer necessary.

File: sync/02-cleanup.sh (hunk 4 above)
Comment (karelmaxa) [reply sync/02-cleanup.sh]: Cleanup has been completely removed.

File: sync/01-sync.sh (hunk 3 above)
Comment (karelmaxa) [reply sync/01-sync.sh]: Yes, I knew it. I have introduced a while loop to check the `audit/sync` logs.

**Review by pavelhoral — APPROVED** [review]
LGTM

**Pull request merged**